src/dmd/dcast.d

```diff
@@ -1929,9 +1929,8 @@ Expression castTo(Expression e, Scope* sc, Type t)
                     for (size_t u = 0; u < e.len;)
                     {
                         dchar c;
-                        const p = utf_decodeChar(se.peekString().ptr, e.len, u, c);
-                        if (p)
-                            e.error("%s", p);
+                        if(const p = utf_decodeChar(se.peekString()[0 .. e.len], u, c))
```

Comment:
`e.len == peekString().length`, so no extra slicing necessary.

Reply:
Please put a space between `if` and `(`. It's not a function call.

Reply:
I try to use `p` for pointers. Please use `s` for strings if you can.